packages/commonwealth/server/routes/moveAddress.ts

```diff
@@ -0,0 +1,78 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { DB } from '../models';
+
+export const Errors = {
+  NotAuthorized: 'Not authorized',
+  InvalidUpdate: 'Invalid update',
+  NoAddressProvided: 'No address provided in query',
+  NoOldProfileProvided: 'No old profile provided in query',
+  NoNewProfileProvided: 'No new profile provided in query',
+};
+
+const moveAddress = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.body.addressHash) {
+    return next(new Error(Errors.NoAddressProvided));
+  }
+
+  if (!req.body.oldProfileId) {
+    return next(new Error(Errors.NoOldProfileProvided));
+  }
+
+  if (!req.body.newProfileId) {
+    return next(new Error(Errors.NoNewProfileProvided));
+  }
+
+  const { addressHash, oldProfileId, newProfileId } = req.body;
+
+  const address = await models.Address.findOne({
+    where: {
+      address: addressHash,
```

Comment:
This is not a hash of the address, it is the address. We should rename to reflect that.